Editor/Overlays/SelectionSettingsButtons.cs

```diff
@@ -0,0 +1,142 @@
+﻿#if UNITY_2023_2_OR_NEWER
+using System;
+using System.Collections.Generic;
+using UnityEditor.Overlays;
+using UnityEditor.ProBuilder.Actions;
+using UnityEditor.Toolbars;
+using UnityEngine;
+using UnityEngine.ProBuilder;
+using UnityEngine.UIElements;
+
+namespace UnityEditor.ProBuilder.UI
+{
+    [EditorToolbarElement("ProBuilder Settings/Drag Rect Mode")]
+    class DragRectModeToggle : EditorToolbarToggle
+    {
+        DragRectModeToggle()
+        {
+            var action =  EditorToolbarLoader.GetInstance<ToggleDragRectMode>();
+            tooltip = action.tooltip.summary;
+            style.display = action.hidden ? DisplayStyle.None : DisplayStyle.Flex;
+
+            offIcon = action.icons[0];
+            onIcon = action.icons[1];
+            SetValueWithoutNotify(ProBuilderEditor.rectSelectMode == RectSelectMode.Complete);
+            RegisterCallback<ChangeEvent<bool>>(evt => ProBuilderEditor.rectSelectMode = evt.newValue ? RectSelectMode.Complete : RectSelectMode.Partial);
+
+            ProBuilderEditor.selectModeChanged += (s) => SelectModeUpdated();
```

Comment:
Is this not being unregistered going to cause issues? If this needs to be unregistered, you can move this to AttachToPanel and unregister during DetachFromPanel.

Reply:
Addressed in [98f445b](https://github.com/Unity-Technologies/com.unity.probuilder/pull/468/commits/98f445b270e264883a6fd80ac90e35713021c8c5)